### PR TITLE
Error message for PAY_PER_REQUEST billing mode

### PR DESCRIPTION
**Pull request by giograno**
As reported in #4982, when a table is created with `PAY_PER_REQUEST` billing mode and a provisioned throughput is specified as well, we should return a `ValidationException`.

### CODE REVIEW

File: localstack/services/dynamodb/provider.py [hunk 1]
```diff
@@ -386,6 +387,15 @@ def create_table(
         table_name = create_table_input["TableName"]
         if self.table_exists(table_name):
             raise ResourceInUseException("Cannot create preexisting table")
+        billing_mode = create_table_input.get("BillingMode", None)
```
Comment (thrau) [inline localstack/services/dynamodb/provider.py:390]: nit: `dict.get("key", None)` is equivalent to `dict.get("key")`

File: tests/integration/test_dynamodb.py [hunk 2]
```diff
@@ -978,6 +978,20 @@ def test_dynamodb_batch_write_item(self):
...
+
+        with pytest.raises(Exception) as e:
+            dynamodb.create_table(
+                TableName=table_name,
+                KeySchema=[{"AttributeName": "id", "KeyType": "HASH"}],
+                AttributeDefinitions=[{"AttributeName": "id", "AttributeType": "S"}],
+                ProvisionedThroughput={"ReadCapacityUnits": 5, "WriteCapacityUnits": 5},
+                BillingMode="PAY_PER_REQUEST",
+            )
+        assert "One or more parameter values were invalid" in str(e.value)
```
Comment (thrau) [inline tests/integration/test_dynamodb.py:993]: nit: we can use the `pytest.raises` context to assert exception messages:  `e.matches("One or ...`

**Review by thrau — APPROVED** [review]
LGTM! just a couple of minor comments, nothing blocking

**Comment by giograno** [comment]
Had to slightly touch the cloudformation model to accomodate for the changes in the provider (cc @dominikschubert )

**Pull request merged**